cpp/src/IceStorm/TransientTopicManagerI.cpp

```diff
@@ -64,12 +42,27 @@ TransientTopicManagerImpl::retrieve(string name, const Ice::Current&)
         throw NoSuchTopic(name);
     }
 
-    // Here we cannot just reconstruct the identity since the
-    // identity could be either instanceName/topic name, or if
-    // created with pre-3.2 IceStorm / topic name.
     return _instance->topicAdapter()->createProxy<TopicPrx>(p->second->id());
 }
 
+optional<TopicPrx>
+TransientTopicManagerImpl::createOrRetrieve(string name, const Ice::Current&)
+{
+    lock_guard lock(_mutex);
+
+    reap();
+
+    auto p = _topics.find(name);
+    if (p != _topics.end())
+    {
+        return _instance->topicAdapter()->createProxy<TopicPrx>(p->second->id());
+    }
+    else
+    {
+        return createImpl(std::move(name));
+    }
```

Comment:
It's better to not include the else when if statement is terminal. 
```suggestion
    auto p = _topics.find(name);
    if (p != _topics.end())
    {
        return _instance->topicAdapter()->createProxy<TopicPrx>(p->second->id());
    }

    return createImpl(std::move(name));
    
```